src/js/utils/buffer.js

```diff
@@ -0,0 +1,35 @@
+import { createTimeRange } from './time-ranges.js';
+
+/**
+ * Compute how much your video has been buffered
+ * @param  {Object} Buffered object
+ * @param  {Number} Total duration
+ * @return {Number} Percent buffered of the total duration
+ * @private
+ */
+export function bufferedPercent(buffered, duration) {
+  var bufferedDuration = 0,
+      start, end;
+
+  if (!duration) {
+    return 0;
+  }
+
+  if (!buffered || !buffered.length) {
+    buffered = createTimeRange(0,0);
+  }
+
+  for (let i=0; i<buffered.length; i++){
```

Comment:
those missing whitespaces were actually original cut/paste of the old code but I'll fix it anyway :)